lib/compress.js

```diff
@@ -774,13 +767,6 @@ merge(Compressor.prototype, {
                 }
             }
 
-            // Remove extraneous empty statments in block after removing var definitions.
-            // Leave at least one statement in `statements`.
```

Comment:
`var self = compressor.self();` above was not guaranteed to be `AST_Scope` (which it assumes and used below in places like `ref.scope !== self`

Since I've patched that up in https://github.com/mishoo/UglifyJS2/pull/1826/commits/1baf3a8f109f6e5630983d35e88a1c3db74e8f98, we can now safely remove this workaround.